shared/bazel/rules/robotpy/semiwrap_helpers.bzl

```diff
@@ -0,0 +1,345 @@
+
+def publish_casters(
+        name,
+        project_config,
+        caster_name,
+        output_json,
+        output_pc,
+        typecasters_srcs,
+        package_root):
+    """
+    Sugar wrapper for the semiwrap.cmd.publish_casters tool
+    """
+    cmd = _wrapper() + " semiwrap.cmd.publish_casters"
+    cmd += " $(SRCS) " + caster_name + " $(OUTS)"
+
+    native.genrule(
+        name = name,
+        srcs = [project_config],
+        outs = [output_json, output_pc],
+        cmd = cmd,
+        tools = _wrapper_dep() + typecasters_srcs + [package_root],
+        visibility = ["//visibility:public"],
+        tags = ["robotpy"],
+    )
+
+def resolve_casters(
+        name,
+        casters_pkl_file,
+        dep_file,
+        caster_files = [],
+        caster_deps = []):
+    """
+    Sugar wrapper for the semiwrap.cmd.resolve_casters tool
+    """
+    cmd = _wrapper() + " semiwrap.cmd.resolve_casters "
+    cmd += " $(OUTS)"
+
+    cmd += _location_helper(_semiwrap_caster()) + "/semiwrap/semiwrap.pybind11.json"
+
+    resolved_caster_files = []
+
+    deps = []
+    for dep in caster_deps:
+        deps.append(dep)
+        cmd += _location_helper(dep)
+
+    for cfd in caster_files:
+        if cfd.startswith(":"):
+            resolved_caster_files.append(cfd)
+            cmd += _location_helper(cfd)
+        else:
+            cmd += " " + cfd
+
+    native.genrule(
+        name = name,
+        srcs = resolved_caster_files + deps,
+        outs = [RESOLVE_CASTERS_DIR + casters_pkl_file, RESOLVE_CASTERS_DIR + dep_file],
+        cmd = cmd,
+        tools = _wrapper_dep() + [_semiwrap_caster()],
+        tags = ["robotpy"],
+    )
+
+def gen_libinit(
+        name,
+        output_file,
+        modules):
+    """
+    Sugar wrapper for the semiwrap.cmd.gen_libinit tool
+    """
+    cmd = _wrapper() + " semiwrap.cmd.gen_libinit "
+    cmd += " $(OUTS) "
+    cmd += " ".join(modules)
+
+    native.genrule(
+        name = name,
+        outs = [output_file],
+        cmd = cmd,
+        tools = _wrapper_dep(),
+        tags = ["robotpy"],
+    )
+
+def gen_pkgconf(
+        name,
+        project_file,
+        module_pkg_name,
+        pkg_name,
+        output_file,
+        libinit_py,
+        install_path,
+        package_root):
+    """
+    Sugar wrapper for the semiwrap.cmd.gen_pkgconf tool
+    """
+    cmd = _wrapper() + " semiwrap.cmd.gen_pkgconf "
+    cmd += " " + module_pkg_name + " " + pkg_name
+    cmd += _location_helper(project_file)
+    cmd += " $(OUTS)"
+    if libinit_py:
+        cmd += " --libinit-py " + libinit_py
+
+    OUT_FILE = install_path + "/" + output_file
+    native.genrule(
+        name = name,
+        srcs = [package_root],
+        outs = [OUT_FILE],
+        cmd = cmd,
+        tools = _wrapper_dep() + [project_file],
+        visibility = ["//visibility:public"],
+        tags = ["robotpy"],
+    )
+
+def header_to_dat(
+        name,
+        casters_pickle,
```

Comment:
These sugar things are a bit weird to me. Doesn't makeplan already emit the arguments needed? Why are they getting re-rendered in these helpers? If the semiwrap tool arguments change for some reason then this will break.